test/server/http/admin_test.cc

```diff
@@ -650,6 +651,38 @@ TEST_P(AdminInstanceTest, TracingStatsDisabled) {
   }
 }
 
+TEST_P(AdminInstanceTest, GetRequest) {
+  Http::HeaderMapImpl response_headers;
+  std::string body;
+  EXPECT_EQ(Http::Code::OK, admin_.request("/server_info", Http::Utility::QueryParams(), "GET",
+                                           response_headers, body));
+  EXPECT_TRUE(absl::StartsWith(body, "envoy ")) << body;
+  EXPECT_THAT(std::string(response_headers.ContentType()->value().getStringView()),
+              HasSubstr("text/plain"));
+}
+
+TEST_P(AdminInstanceTest, GetRequestJson) {
+  Http::HeaderMapImpl response_headers;
+  std::string body;
+  EXPECT_EQ(Http::Code::OK,
+            admin_.request("/stats", Http::Utility::QueryParams({{"format", "json"}}), "GET",
```

Comment:
Probably should have a test for > 1 query param, since there is slightly different logic (switches to & separator).

Reply:
I couldn't find an admin handler that dealt with >1 query param, but I factored out the serialization to an http utility function and added tests for that to cover all the cases.